### ISSUE TITLE
Tainted syntax trying to debug minimal typed/racket program

### ISSUE THREAD

**Issue by greghendershott** [racket/drracket#231]
1. Open Dr Racket 7.0.
2. Change _Definitions_ simply to `#lang typed/racket` (plus newline)
3. Press the `Debug` button.

Error prints in _Interactions_: `../../Applications/Racket_v7.0/collects/racket/private/pre-base.rkt:182:64: ?: cannot bind from tainted syntax in: (quote #%kernel)`.

(This seems to be an old issue; similar error in Racket versions 6.5, 6.7, and 6.10: `../../Applications/Racket_v6.10/collects/racket/private/pre-base.rkt:207:39: module: cannot use identifier tainted by macro transformation in: module`.)

Interestingly this is the same location I referenced in #230. Adding the test I asked about there, does not affect this bug either way.

**Comment by greghendershott** [comment]
The following change to [`module-annotate` in `drracket/gui-debugger/annotator.rk`](https://github.com/racket/drracket/blob/master/drracket/gui-debugger/annotator.rkt#L166-L179) _seems_ to fix the issue for me (note the "before" diff is already reformatted for line width compared to the original):

```diff
--- before.rkt	2018-08-22 14:29:16.000000000 -0400
+++ after.rkt	2018-08-22 14:30:11.000000000 -0400
@@ -1,17 +1,18 @@
 (define (module-annotate stx)
   (syntax-case stx ()
     [(_ identifier name mb)
      (syntax-case (disarm #'mb) ()
        [(plain-module-begin . module-level-exprs)
         (with-syntax ([(module . _) stx])
           (quasisyntax/loc stx
             (module identifier name
               #,(rearm
                  #'mb
                  #`(plain-module-begin
                     #,@(map (lambda (e)
-                              (module-level-expr-iterator
-                               e
-                               (list (syntax-e #'identifier)
-                                     (syntax-source #'identifier))))
+                              (rearm
+                               (module-level-expr-iterator
+                                (disarm e)
+                                (list (syntax-e #'identifier)
+                                      (syntax-source #'identifier)))))
                             (syntax->list #'module-level-exprs)))))))])]))
```

In other words, simply `disarm` and `rearm` each module level expression.

I thought to try this from grepping for `syntax-arm` in Typed Racket and finding [this in `utils/arm.rkt`](https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/utils/arm.rkt#L12-L14):

```racket
;; For simplicity, protect everything produced by Typed Racket.
(define (arm stx)
  (syntax-case stx (#%plain-module-begin
                    #%require #%provide #%declare begin
                    define-values define-syntaxes
                    begin-for-syntax)
    [(#%plain-module-begin . _) (syntax-property (syntax-arm stx)
                                                 'taint-mode
                                                 'opaque)]
...
```

However, I barely understand syntax taints. This is my first hands-on use. Is it this simple??

**Comment by wanderley** [comment]
I am hitting the same issue in Racket 7.1 and Racket 7.2.

**Comment by Nyanraltotlapun** [comment]
I am having the same issue with Racket 7.3

**Comment by xinyuzheng7** [comment]
I am hitting the same issue in Racket 7.4 with sicp collection..

**Comment by PrintScr** [comment]
Same issue ...

**Comment by greghendershott** [comment]
Why I opened this issue instead of submitting a pull request:

I had looked at the error message, found some code in Typed Racket that seemed relevant (??), blindly copied it, and hoped "it worked".

I imagined someone who _actually_ understands syntax taints -- such as @rfindler @samth @rmculpepper @mflatt -- could take a look and say one of:

- "Unfortunately it's not that simple: $WHY_AND_WHAT_TO_DO_INSTEAD"
- "LGTM, please submit a PR"
- "Heck it's faster just to push such a commit myself" :smile:

**Comment by sorawee** [comment]
@soegaard asked this in Slack today, and @mflatt said it's already fixed and will be released in Racket 7.6. I also just ran (pre-release) Racket 7.6 and confirmed that it's indeed fixed. Good to close, @greghendershott?

**Comment by greghendershott** [comment]
Great! That looks like commit 82f514f. Closing.

**Comment by k4rtik** [comment]
Ah, so this is fixed in 7.6. I was doing a live demo for the debugger in class today and ran into this issue with v7.5, I had only tested this on v7.6 on a different machine.

Thanks for helping push it forward @greghendershott. :)
